File: backend/src/main/java/com/kerflowapp/kerflow/api/users/domain/ApiTokenDto.java
package com.kerflowapp.kerflow.api.users.domain;

import lombok.Builder;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record ApiTokenDto(
    UUID id,
    String name,
    String tokenPrefix,
    LocalDateTime creationDate,
    Instant lastUsedAt,
    Instant revokedAt
) {
}

File: frontend/src/api/dtos/api-token.dto.ts
export interface ApiTokenDto {
	id: string;
	name: string;
	tokenPrefix: string;
	creationDate: string;
	lastUsedAt?: string;
	revokedAt?: string;
}

export interface CreatedApiTokenDto {
	id: string;
	name: string;
	tokenPrefix: string;
	token: string;
}
